### ISSUE TITLE
Error opening a Realm with Realm.10.6.0 - c# .NET

### ISSUE THREAD

**Issue by KetiT** [realm/realm-dotnet#2673]
I have Windows forms app in C# that was working perfectly fine with older Realm library 3.4.0. Now, I updated to Realm 10.6.0 through NuGet console and all packages and libraries were installed fine, no errors during compiling. I'm using .Net 4.7.2 and Visual Studio 2019. I have FodyWeaver.xml in the project included. But, i get error when i try to open and read RealmConfiguration from a realm file.

config = new RealmConfiguration(dbPath);                 
realm = Realm.GetInstance(config);
var contacts = realm.All<RealmClasses.RealmContact>();

I can open the realm in MongoRealmDBStudio without problems. 
However, from the code, i get an error on third line: "The class RealmContact is not in the limited set of classes for this realm". Config object has values null for ObjectClasses and Schema.

Even if i try to create the default configuration and load classes manually, i get similar error "No properties in RealmContact, has linker stripped it?" I also get ""No RealmObjects. Has linker stripped them?" error sometimes.

Realm classes are generated from the realm file itself using latest version of MongoDB Realm Studio.

I even downloaded and tried to open the Demo realm file that comes with MongoDB Realm Studio - same errors.

I cleaned up the Solution, manually deleted bin and obj folders, even created new solution/project and added all packaged again, didn't work.

How can i open Realm file without issues and read its Realm configuration, what am i missing? 

![RealmContactClass](https://user-images.githubusercontent.com/43048547/135922488-9b1b9733-390f-4b27-9198-e939561f546a.png)
![RealmFile](https://user-images.githubusercontent.com/43048547/135922491-7fe3b620-1866-43f2-9a83-8364df8ff7c7.png)

**Comment by nirinchev** [comment]
Hm, if you're upgrading from 3.x, can you double check that your `FodyWeavers.xml` contains `<Realm />` rather than `<RealmWeaver />`?

**Comment by KetiT** [comment]
Yes, this is changed to <Realm />

**Comment by nirinchev** [comment]
Hm... this definitely sounds like you're hitting some corner case we haven't covered well. Do you think you can share your project with us so that we can run it locally and try to reproduce the problem?

**Comment by KetiT** [comment]
Yes, here is the modified project. Realm file is there too. I didn't include the packages folder in the zip file, but here is the screenshot:

![image](https://user-images.githubusercontent.com/43048547/136057459-41d31054-0cf5-4b43-a8a7-f13eacfb4531.png)

[TestRealm.zip](https://github.com/realm/realm-dotnet/files/7287258/TestRealm.zip)

**Comment by nirinchev** [comment]
Okay, I can see what the issue is, but I'm not sure how this has ever worked. It might be something that went wrong during the migration, but your `RealmConnection` project didn't reference the Realm nuget package correctly - it was just pulling in the `Realm.dll` binary, but was not referencing our other dependencies, including Fody, which is actually the package that goes and converts your C# models into Realm-compatible classes through a process known as code weaving. To fix the issue, I had to remove the `Realm.dll` reference from `RealmConnection` and add the nuget package so that it includes all dependencies when compiling the project. I'm uploading the fixed solution for your reference - let me know if that works for your main project. Generally, the rule of thumb is to just include the Realm package in all projects that contain model classes and your main executable project. 

[TestProjectForRealm.zip](https://github.com/realm/realm-dotnet/files/7288402/TestProjectForRealm.zip)
